src/main/java/com/microsoft/aad/adal4j/AdalDeviceCodeAuthorizationGrant.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright © Microsoft Open Technologies, Inc.
+ *
+ * All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+ * OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+ * ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
+ * PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ * See the Apache License, Version 2.0 for the specific language
+ * governing permissions and limitations under the License.
+ ******************************************************************************/
+package com.microsoft.aad.adal4j;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Class for device code grant.
+ */
+public class AdalDeviceCodeAuthorizationGrant implements AdalAuthorizationGrant {
```

Comment:
Is the AdalAuthorizationGrant interface a representation of the OAuth grant types? If so, why is the AdalDeviceCodeAuthorizationGrant  a separate implementation from the AdalOAuthAuthorizationGrant class? The device code flow is also one of the OAuth grant types as per this [spec](https://tools.ietf.org/html/draft-ietf-oauth-device-flow-08), right?

Reply:
according to Nimbus sdk it is not